pyproject.toml

```diff
@@ -2,6 +2,12 @@
 build-backend = "hatchling.build"
 requires = ["hatch-vcs", "hatchling"]
 
+[dependency-groups]
+dev = [
+  "mne[dev]",
```

Comment:
I believe to really get *everything* one needs to do `mne[full-pyside6,test-extra,dev]` (or sub `full-pyside6` for `full` to get PyQt6)

Reply:
Just like with extras, we can add as many dependency groups as we like. I'd say that really everything should probably not be assigned to `dev`, but maybe `dev-full` or something like that. A more lightweight dev setup should probably also be supported by the `test` dependency group.

Regarding the warning on duplicate extras and dependency group names, uv does not have any issue with it. pip does not support dependency groups AFAIK, so we should be good.